Fix miscellaneous compiler warnings, July 2022 edition

mcatanzaro opened the pull request:
#### aff87f6cbcf56b5fe448770a00f51814fa626b7c
<pre>
Fix miscellaneous compiler warnings, July 2022 edition
<a href="https://bugs.webkit.org/show_bug.cgi?id=243065">https://bugs.webkit.org/show_bug.cgi?id=243065</a>

Reviewed by Chris Dumez.

* Source/WebCore/accessibility/atspi/AccessibilityObjectAtspi.cpp:
(WebCore::atspiRole):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::getAttributeForBindings const):

Canonical link: <a href="https://commits.webkit.org/252745@main">https://commits.webkit.org/252745@main</a>
</pre>

cdumez commented on Source/WebCore/accessibility/atspi/AccessibilityObjectAtspi.cpp:
```diff
@@ -385,10 +385,10 @@ static unsigned atspiRole(AccessibilityRole role)
-        return Atspi::Role::Unknown;
     case AccessibilityRole::ListMarker:
     case AccessibilityRole::MathElement:
-        RELEASE_ASSERT_NOT_REACHED();
```
Looks like a behavior change?

mcatanzaro replied:
Yes, I suppose I should mention that in the commit message.

cdumez replied:
Bu why are we changing behavior when trying to silence a warning. This doesn't seem right.

mcatanzaro replied:
Because I wanted to be lazy and not solve it in a separate commit. I can split it into a second PR if you prefer. Crashing seems like overkill here, and there is already a big list of unknown accessibility roles, so using that seems appropriate.

mcatanzaro replied:
I'll use just this one PR unless you request it be split into two.

Note the title is "fix compiler warnings," not "silence compiler warnings," so behavior changes are probably not completely disallowed.

cdumez replied:
Atspi isn't used by Cocoa ports, right? If so, do your thing.

mcatanzaro replied:
Yes, this is the Linux accessibility technology.

carlosgcampos replied:
Have you seen a crash because of this? At least in some of the cases, like ListMarker, I think it's actually imposible to get that role at this point, because they are specially handled.

mcatanzaro replied:
Nope, I just assumed there was a mistake for some roles to be marked as Unknown and for others to trigger a crash. I'll change it back then, and add a warning comment.

I assume new roles should be marked as Unknown, right?

cdumez commented on Source/WebCore/dom/Element.cpp:
```diff
@@ -733,7 +733,7 @@ AtomString Element::getAttributeForBindings(const QualifiedName& name, ResolveUR
     }
 
-    ASSERT_NOT_REACHED();
+    RELEASE_ASSERT_NOT_REACHED();
```
Can we write this differently so it silences the warning but doesn't add release time code to this hot function?

mcatanzaro replied:
We could use IGNORE_RETURN_TYPE_WARNINGS_BEGIN/END

cdumez commented on Source/WebCore/dom/Element.cpp:
```diff
@@ -1831,7 +1831,7 @@ AtomString Element::getAttributeForBindings(const AtomString& qualifiedName, Res
     }
 
-    ASSERT_NOT_REACHED();
+    RELEASE_ASSERT_NOT_REACHED();
```
ditto.

cdumez commented on Source/WebCore/dom/Element.cpp:
```diff
@@ -734,7 +734,10 @@ AtomString Element::getAttributeForBindings(const QualifiedName& name, ResolveUR
 
     ASSERT_NOT_REACHED();
+
+IGNORE_RETURN_TYPE_WARNINGS_BEGIN
```
Couldn't we solve this without the need for such macros? For example:
```
case ResolveURLs::No:
    break;
}
return attribute->value();
```

I'd really like it if we avoid reduce the spread of such macros to silence warnings.

mcatanzaro replied:
Well it's usually not that easy, or not so nice... but in this case, yes, sure.

cdumez reviewed (approved):
LGTM

mcatanzaro committed:
Fix miscellaneous compiler warnings, July 2022 edition
https://bugs.webkit.org/show_bug.cgi?id=243065

Reviewed by Chris Dumez.

* Source/WebCore/accessibility/atspi/AccessibilityObjectAtspi.cpp:
(WebCore::atspiRole):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::getAttributeForBindings const):

Canonical link: https://commits.webkit.org/252745@main
